learning_journal/views/views.py

```diff
@@ -0,0 +1,45 @@
+from pyramid.view import view_config
+from pyramid.response import Response
+from sqlalchemy.exc import DBAPIError
+from ..models import Entry
+
+DB_ERROR = "Whoops, there was a problem with the database!"
+
+
+@view_config(route_name='index', renderer='../templates/index.jinja2')
+def index_view(request):
+    try:
+        all_entries = request.dbsession.query(Entry).order_by(Entry.id.desc())
+    except DBAPIError:
+        return Response(DB_ERROR, content_type="text/plain", status=500)
+    return {'all_entries': all_entries}
+
+
+@view_config(route_name='entry_details',
+             renderer='../templates/entry_details.jinja2')
+def entry_details(request):
+    try:
+        entry = request.dbsession.query(Entry).get(request.matchdict["id"])
```

Comment:
It is possible, given the way that `.get(id)` works, that the return value might be `None`.  What happens to your view here when that's the case?  What do you think should happen?